caluma/form/serializers.py

```diff
@@ -644,3 +644,19 @@ def update(self, instance, validated_data):
     class Meta:
         fields = ("answer",)
         model = models.Answer
+
+
+class RemoveDocumentSerializer(serializers.ModelSerializer):
+    document = serializers.GlobalIDField(source="id")
+
+    def update(self, instance, validated_data):
+        if hasattr(instance, "case"):
+            raise Exception("You cannot remove a Document, if it's attached to a case.")
+
+        instance.delete()
```

Comment:
From the discussion - shouldn't we recursively delete subdocuments if the document contains table answers?

If it makes sense, maybe parametrize the mutation so the client can select what to do in that case

Reply:
Good catch! To me it seems most consistent to also remove subdocuments.

Reply:
It's implemented. Without a parameter for now.

Reply:
Thanks! However this should probably be recursive. I know nobody does nested tables at the moment, but one day people might do it and get a surprising behaviour..

Reply:
AFAIK the consensus is to NOT support nested tables. @czosel

Reply:
Fair enough, let's just not forget that :)

Reply:
Right - I think we should discuss this in person in more detail once. On the one hand we consciously decided to avoid the complexity involved in nested tables for the moment (both front- and backend), on the other hand we do have a (non-caluma) project where is is currently a requirement. It might be worth discussing what kind of alternative designs might be possible, so that we're prepared for the day where this becomes important :wink: